spec/models/ip_spec.rb

```diff
@@ -1,51 +1,56 @@
 describe Ip do
+  subject(:ip_address) { described_class.new }
+
   it { is_expected.to belong_to(:location) }
+  it { is_expected.to validate_presence_of(:address) }
+  it { is_expected.to validate_uniqueness_of(:address) }
+
+  context "when validating" do
+    let(:location) { create(:location, organisation: create(:organisation)) }
+
+    it 'does not allow the address 0.0.0.0' do
+      ip = described_class.create(address: '0.0.0.0', location: location)
+      expect(ip.errors.full_messages).to eq([
+        "Address '0.0.0.0' is not valid"
+      ])
+    end
+
+    context "with an invalid address" do
+      let!(:ip) { described_class.create(address: "invalidIP", location: location) }
 
-  context "validations" do
-    it { is_expected.to validate_presence_of(:address) }
-    it { is_expected.to validate_uniqueness_of(:address) }
-
-    context "address" do
-      let(:location) { create(:location, organisation: create(:organisation)) }
-
-      context "when invalid" do
-        let!(:ip) { described_class.create(address: "invalidIP", location: location) }
-
-        it "does not save when address is an invalid IP" do
-          expect(described_class.count).to eq(0)
-          expect(ip.errors.full_messages).to eq([
-            "Address 'invalidIP' is not valid"
-          ])
-        end
-
-        it 'prevents 0.0.0.0' do
-          ip = described_class.create(address: '0.0.0.0', location: location)
-          expect(ip.errors.full_messages).to eq([
-            "Address '0.0.0.0' is not valid"
-          ])
-        end
+      it "does not save" do
+        expect(described_class.count).to eq(0)
       end
 
-      context "when valid" do
-        let!(:ip) { described_class.create(address: "141.0.149.130", location: location) }
+      it "displays an error message" do
+        expect(ip.errors.full_messages).to eq([
+          "Address 'invalidIP' is not valid"
+        ])
+      end
+    end
+
+    context "with a valid address" do
+      let!(:ip) { described_class.create(address: "141.0.149.130", location: location) }
+
+      it "saves the address" do
```

Comment:
Actually it saves the `Ip`, not the `address`. `address` is an attribute on the `Ip` model.